agent/agent.go

```diff
@@ -887,8 +887,10 @@ func (a *Agent) makeServer() (*consul.Server, error) {
 	if err != nil {
 		return nil, err
 	}
-	if err := a.setupKeyrings(config); err != nil {
-		return nil, fmt.Errorf("Failed to configure keyring: %v", err)
+	if !a.config.DisableKeyringFile {
```

Comment:
This isn't a bad idea since we are creating the config and not worrying about BC. If we switch to a `bool*` we could make this `EnableKeyringFile` and default it to `true`.